sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -355,18 +362,44 @@ def _create_handler(self, auth):
         )
         if self._prefetch_count == 1:
             self._handler._message_received = self._enhanced_message_received  # pylint: disable=protected-access
+    
+    def _create_auth(self):
+        # type: () -> JWTTokenAuth
+        """
+        Create an ~uamqp.authentication.SASTokenAuth instance to authenticate
+        the session.
+        """
+        try:
+            # ignore mypy's warning because token_type is Optional
+            token_type = self._credential.token_type  # type: ignore
+        except AttributeError:
+            token_type = b"jwt"
+        if token_type == b"servicebus.windows.net:sastoken":
+            return JWTTokenAuth(
+                self._auth_uri,
+                self._auth_uri,
+                functools.partial(self._credential.get_token, self._auth_uri)
+            )
+        return JWTTokenAuth(
+            self._auth_uri,
+            self._auth_uri,
+            functools.partial(self._credential.get_token, JWT_TOKEN_SCOPE),
+            token_type=token_type,
+            timeout=self._config.auth_timeout,
+            custom_endpoint_hostname=self._config.custom_endpoint_hostname,
+            port=self._config.connection_port,
+            verify=self._config.connection_verify,
+        )
 
     def _open(self):
         # pylint: disable=protected-access
-        if self._running:
-            return
-        if self._handler and not self._handler._shutdown:
-            self._handler.close()
-
-        auth = None if self._connection else create_authentication(self)
+        if not self._running:
+            if self._handler:
+                self._handler.close()
+        auth = self._create_auth()
         self._create_handler(auth)
         try:
-            self._handler.open(connection=self._connection)
+            self._handler.open()
```

Comment:
Why are we no longer passing in the connection? Does this happen somewhere else now?